parse/parser.go

```diff
@@ -534,7 +605,7 @@ func parseIfStatement(buf TokenBuffer) (*ast.IfStatement, error) {
 	}
 
 	tok = buf.Read()
-	if curTokenIs(buf, Else) {
+	if err := expectNext(buf, Else); err == nil {
```

Comment:
else is optional in if-else statements so I think it is better not to use `expect` function